streaming/src/test/scala/org/apache/spark/streaming/scheduler/SchedulerSuite.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.streaming.scheduler
+
+import org.apache.spark.streaming._
+import org.apache.spark.SparkConf
+import org.apache.spark.storage.StorageLevel
+import org.apache.spark.streaming.receiver._
+import org.apache.spark.util.Utils
+
+/** Testsuite for receiver scheduling */
+class SchedulerSuite extends TestSuiteBase {
+  val sparkConf = new SparkConf().setMaster("local[8]").setAppName("test")
+  val ssc = new StreamingContext(sparkConf, Milliseconds(100))
+  val tracker = new ReceiverTracker(ssc)
+  val launcher = new tracker.ReceiverLauncher()
+
+  test("receiver scheduling - no preferredLocation") {
+    val numReceivers = 10;
+    val receivers = (1 to numReceivers).map(i => new DummyReceiver)
+    val executors: List[String] = List("Host1", "Host2", "Host3", "Host4", "Host5")
+    val locations = launcher.scheduleReceivers(receivers, executors)
+    assert(locations(0)(0) === "Host1")
+    assert(locations(4)(0) === "Host5")
+    assert(locations(5)(0) === "Host1")
+    assert(locations(9)(0) === "Host5")
+  }
+
+  test("receiver scheduling - no preferredLocation, numExecutors > numReceivers") {
+    val numReceivers = 3;
+    val receivers = (1 to numReceivers).map(i => new DummyReceiver)
```

Comment:
Structure of the four tests, which could evolve separately as opposed to being centralized. The way these tests are written is pretty consistent with the rest of spark test suites, which can be easily read and debugged independently at the cost of slight code duplication. Take a look at JobCancellationSuite for example. What you're suggesting is elegant but adds complexity and makes the code hard to read. If we had a dozen such tests, the verbosity would bother me too. Will try and reduce the number of asserts, I think some of them are redundant.